deploy/osps/default/osp-flatcar.yaml

```diff
@@ -51,7 +51,56 @@ spec:
         {{- end }}
 
         source /etc/environment
-
+      configureVCloudNetwork: |-
+        function query_ovf () {
+          local QUERY=$1
+          echo $(/usr/share/oem/bin/vmtoolsd --cmd "info-get guestinfo.ovfEnv" | grep "${QUERY}" | sed -e 's/.*oe:value="//g' | sed 's/"\/>//g')
+        }
+        
+        mask2cdr () {
+          local mask=$1
+        
+          # In RFC 4632 netmasks there's no "255." after a non-255 byte in the mask
+          local left_stripped_mask=${mask##*255.}
+          local len_mask=${#mask}
+          local len_left_stripped_mask=${#left_stripped_mask}
+        
+          local conversion_table=0^^^128^192^224^240^248^252^254^
+          local number_of_bits_stripped=$(( ($len_mask - $len_left_stripped_mask)*2 ))
+          local signifacant_octet=${left_stripped_mask%%.*}
+        
+          local right_stripped_conversion_table=${conversion_table%%$signifacant_octet*}
+          local len_right_stripped_conversion_table=${#right_stripped_conversion_table}
+          local number_of_bits_from_conversion_table=$((len_right_stripped_conversion_table/4))
+          echo $(( $number_of_bits_stripped + $number_of_bits_from_conversion_table ))
+        }
+        
+        HOSTNAME=$(query_ovf "vCloud_computerName")
+        PRIMARY_NIC=$(query_ovf "vCloud_primaryNic")
+        IP_MODE=$(query_ovf "vCloud_bootproto_${PRIMARY_NIC}")
+        
+        if [ "$IP_MODE" = "static" ]; then
+        IP=$(query_ovf "vCloud_ip_${PRIMARY_NIC}")
+        GATEWAY=$(query_ovf "vCloud_gateway_${PRIMARY_NIC}")
+        DNS1=$(query_ovf "vCloud_dns1_${PRIMARY_NIC}")
+        DNS2=$(query_ovf "vCloud_dns2_${PRIMARY_NIC}")
+        NETMASK=$(query_ovf "vCloud_netmask_${PRIMARY_NIC}")
+        CIDR=$(mask2cdr "$NETMASK")
+        cat <<EOF > /etc/systemd/network/00-ens192.network
+        [Match]
+        Name=ens192
+        
+        [Network]
+        Address=$IP/$CIDR
+        Gateway=$GATEWAY
+        DNS=$DNS1 $DNS2
+        EOF
+        
+        fi
```

Comment:
We need to move this "if" condition to https://github.com/kubermatic/operating-system-manager/pull/330/files#diff-25902bf57716237276474ac08e94b4edd95cfc3c67553c8446c3206df4cb9050R142

Since we don't need to execute these scripts if IP mode is not static.

Reply:
No, you'll have to expose it from [here](https://github.com/kubermatic/operating-system-manager/blob/main/pkg/controllers/osc/resources/operating_system_config.go#L268-L294).

Alternatively, if you can update the condition to make sure that `systemctl restart systemd-networkd` is inside the if block, I'm fine with that as well.

Reply:
Yeah okay im just gonna put it inside the if block 😆